wp-content/plugins/wporg-learn/inc/sensei.php

```diff
@@ -258,3 +260,58 @@ function get_my_courses_page_url() {
 
 	return get_permalink( $page_id );
 }
+
+/**
+ * Modify the course query to add the featured course meta query if set.
+ *
+ * @param mixed $pre_render The pre-render value.
+ * @param mixed $parsed_block The parsed block value.
+ * @return mixed The modified course query.
+ */
+function modify_course_query( $pre_render, $parsed_block ) {
+	if ( isset( $parsed_block['attrs']['namespace'] ) && 'wporg-learn/course-grid' === $parsed_block['attrs']['namespace']
+	) {
+		add_filter(
+			'query_loop_block_query_vars',
+			function( $query, $block ) use ( $parsed_block ) {
+				if ( 'course' !== $query['post_type'] ) {
+					return $query;
+				}
+
+				$course_featured = $parsed_block['attrs']['query']['courseFeatured'];
```

Comment:
is ['query'] guaranteed to be set?

Reply:
Good question

Reply:
Added a guard in https://github.com/WordPress/Learn/pull/2457/commits/8b4f56931f5ac7804b54453d3317eb5da801847e

Reply:
In testing I found that if `$parsed_block['attrs']['query']` was not set, `$course_featured` was `null`. Better safe than sorry anyway.